### DISCUSSION TITLE
Extract MP4 Video from GLTF Viewer

### DISCUSSION THREAD

**Discussion by yogesh895** [google/filament#6446] [Q&A]
Hi, 
Can we extract the MP4 Video from the GLB with animations, textures and lights, directly from the `message.buffer` in [`loadGlb()`](https://github.com/google/filament/blob/ce407ba84487b57262665173ff5dda4a753a0f7a/android/samples/sample-gltf-viewer/src/main/java/com/google/android/filament/gltf/MainActivity.kt#L187) ?

**Comment by romainguy** [comment]
GLBs don't contain MP4 videos. Can you describe what you are trying to do in more detail?

**Reply by yogesh895** [reply]
Hi @romainguy, 
We have GLB containing animations and scenes,  currently we are rendering  it in Texture View on our Application with filament as well as on the Camera Scene with AR Core, our goal is to Generate MP4 Video of the whole Experience.
Are there any possible ways to approach this problem using filament?
Thanks

**Reply by romainguy** [reply]
Yes that is possible but Filament itself isn't involved (at least not directly). You need to use a MediaRecorder which will give you a Surface that you can render into using Filament. You can also look at MediaProjection.
